server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -794,6 +839,7 @@ public ClusterState build() {
                 metadata,
                 routingTable,
                 nodes,
+                Map.copyOf(transportVersions),
```

Comment:
The builder owns this hash map. Can we just wrap it with Collections.unmodifiableMap so a copy is not necessary?